demo/src/main/java/com/novoda/storagepathfinder/demo/StoragePathViewHolder.kt

```diff
@@ -0,0 +1,15 @@
+package com.novoda.storagepathfinder.demo
+
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import com.novoda.storagepathfinder.StoragePath
+import kotlinx.android.synthetic.main.include_device_storage_root.view.*
+
+internal class StoragePathViewHolder(view: View) : RecyclerView.ViewHolder(view) {
```

Comment:
When using kotlin extensions' synthetics make sure the view holder extends `LayoutContainer` otherwise it does a lookup each time:
```
class StoragePathViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: StoragePath, listener: Listener<StoragePath>) {
        device_storage_root.text = item.pathAsString
        add_file.setOnClickListener { listener.invoke(item) }
    }
}
```
It also means that there is no need to reference `itemView` as a nullable. (I would also sort out the names :D )